From 77730c739816427bb9e775479c1ae5729c71580b Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Thu, 9 Oct 2014 21:00:09 -0400
Subject: [fs] xfs: fix incorrect error sign in xfs_file_aio_read

Message-id: <1412888441-4181-16-git-send-email-sandeen@redhat.com>
Patchwork-id: 97320
O-Subject: [RHEL7.1 PATCH 15/47] [fs] xfs: fix incorrect error sign in xfs_file_aio_read
Bugzilla: 1145837
RH-Acked-by: Brian Foster <bfoster@redhat.com>
RH-Acked-by: Dave Chinner <dchinner@redhat.com>

Bugzilla: 1145837
Upstream Status: Committed upstream
Build Info: https://brewweb.devel.redhat.com/taskinfo?taskID=8082426
Tested: locally using xfstests

Thanks,
-Eric

    commit 8ff1e6705a4c711247708f23881feea169e6fc3c
    Author: Dave Chinner <dchinner@redhat.com>
    Date:   Thu May 15 09:21:37 2014 +1000

    xfs: fix incorrect error sign in xfs_file_aio_read

    Signed-off-by: Dave Chinner <dchinner@redhat.com>
    Reviewed-by: Jie Liu <jeff.liu@oracle.com>
    Signed-off-by: Dave Chinner <david@fromorbit.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/xfs/xfs_file.c b/fs/xfs/xfs_file.c
index 3a8e4b6..6a862bb 100644
--- a/fs/xfs/xfs_file.c
+++ b/fs/xfs/xfs_file.c
@@ -295,7 +295,7 @@ xfs_file_aio_read(
   xfs_rw_ilock(ip, XFS_IOLOCK_EXCL);
 
   if (inode->i_mapping->nrpages) {
-   ret = -filemap_write_and_wait_range(
+   ret = filemap_write_and_wait_range(
        VFS_I(ip)->i_mapping,
        pos, -1);
    if (ret) {
-- 
1.7.1